src/mon/MonClient.cc

```diff
@@ -854,6 +857,29 @@ bool MonClient::ms_handle_reset(Connection *con)
   }
 }
 
+bool MonClient::ms_handle_throttle(ms_throttle_t ttype) {
+  switch (ttype) {
+  case ms_throttle_t::MESSAGE:
+    break; // TODO
+  case ms_throttle_t::BYTES:
+    break; // TODO
+  case ms_throttle_t::DISPATCH_QUEUE:
+    {
+      //cluster log a warning that Dispatch Queue Throttle Limit hit
+      if (!log_client) {
+        return false; //cannot handle if the daemon didn't setup a log_client for me
+      }
+      LogChannelRef clog = log_client->create_channel(CLOG_CHANNEL_CLUSTER);
+      clog->warn() << "Throttler Limit has been hit. "
+                   << "Some message processing may be significantly delayed.";
```

Comment:
Let's see how we can make this message more useful. What's the queue depth (# of messages throttled)? What's the bytes in use by the Messenger?

Because of my other suggestion to call this method everytime the throttle is hit, can we also keep track of that? And only send a cluster log warning with a configurable frequency (separate from the messenger debug log config).

Finally, the cluster warning is good but we also want a persistent health check to be generated (only for Ceph daemons, not clients) that shows up in `ceph status`. That will require adding a health metrics to be sent off to the ceph-mgr (after some cursory code reading). You can look at `OSD::get_health_metrics` and its call `mgrc.update_daemon_health(get_health_metrics());` to see how that's done. The challenge is that you can't have the `MonClient` talk directly to the the `MgrClient` due to a circular dependency issue. So the daemon holding the `MonClient` (e.g. `class OSD`) would need to ask it to fill in health checks before forwarding it to the `MgrClient`. Does that make sense?